api/agent/call.go

```diff
@@ -50,12 +50,12 @@ type Params []Param
 
 func FromRequest(appName, path string, req *http.Request) CallOpt {
 	return func(a *agent, c *call) error {
-		app, err := a.da.GetApp(req.Context(), appName)
+		// unfortunately to keep API consistent we have to query route with app name
+		app, err := a.da.GetAppByName(req.Context(), appName)
```

Comment:
i think we have to change this to id for the same reason as the cache / auth / hybrid issue. anything that sits in front of this will be working on a potentially different app with the delete-recreate race.